### PR TITLE
add ability to specify initial values for inputs

### PR DESCRIPTION
**Pull request by zdevito**
Code to export our weights in Toffee IR. Allows the first N arguments to be replaced with N tensors passed in initializers list.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -44,6 +44,14 @@ def test_export_view(self):
         trace, _ = torch.jit.record_trace(lambda x: x.view(1, 1), x)
         self.assertExpected(torch._C._jit_pass_export(trace))
 
+    @toffee_only
+    def test_export_data(self):
+        x = Variable(torch.Tensor([[1, 2], [3, 4]]), requires_grad=True)
+        y = Variable(torch.Tensor([[1, 2], [3, 4]]), requires_grad=True)
+        trace, _ = torch.jit.record_trace(lambda x, y: -torch.sigmoid(torch.tanh(x * (x + y))), x, y)
+        initializers = [x.data]
```
Comment (ezyang) [inline test/test_jit.py:52]: An alternative to passing a tensor and expecting things to positionally match up, is to pass the Variable and infer which input it is using the Variable-to-node mapping. This is a little less flexible (you can't take a Variable and pass a different thing than var.data), but (1) you could always extend the API to be pairs of Variable/Tensor things, so the expressiveness loss is not fundamental, and (2) var.data is basically what you always want here.

File: torch/csrc/toffee/export.cpp [hunk 2]
```diff
@@ -30,6 +30,24 @@ std::string ExportGraph(std::unique_ptr<Graph>& g) {
...
+  for (auto & tensor : initializers) {
+    std::string name = p_g.input(inputs_count++);
+    auto p = p_g.add_initializer();
+    p->set_name(name);
+    for(auto d : tensor.sizes()) {
+      p->add_dims(d);
+    }
+    p->set_data_type(toffee::TensorProto_DataType_FLOAT);
+    //TODO: other types, we force conversion here
+    at::Tensor cont = tensor.toType(at::CPU(at::kFloat));
```
Comment (ezyang) [inline torch/csrc/toffee/export.cpp:44]: Let's fix this

**Comment by ezyang** [comment]
LGTM, but I am a bit skeptical about passing initializers positionally.

**Comment by zdevito** [comment]
Yeah, I want to have a map as well, but I don't know how to do it easily. For now I'd like to get this merged (once ToffeeIR is up to date) and then follow up by improving this interface after it is running end-to-end. 

How should we improve it? The idea to have map rather than a list is the right solution. But the recorded Graph does not keep variable mappings around and I don't think it should because that would couple the Autograph to the Graph representation. However, maybe we should have record_trace return a python dict mapping Variable -> trace_input_idx so that the mapping is preserved along side the graph.

File: test/expect/TestJit.test_export_data.expect [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  attribute {
+    name: "scale"
+    f: -1
+  }
+}
+name: "torch-jit-export"
+input: "1"
+input: "2"
+output: "12"
+initializer {
```
Comment (jerryzh168) [inline test/expect/TestJit.test_export_data.expect:36]: Did you miss an initializer for input "2"?

Comment (ezyang) [reply test/expect/TestJit.test_export_data.expect:36]: Nope. Lack of initializer means that it should be passed in on the other end. E.g., the data you actually want to do inference on.

**Comment by ezyang** [comment]
Merged.
